topgun/k8s/external_postgres_test.go

```diff
@@ -0,0 +1,56 @@
+package k8s_test
+
+import (
+	"fmt"
+	"path"
+
+	. "github.com/concourse/concourse/topgun"
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = Describe("External PostgreSQL", func() {
+	var (
+		releaseName   string
+		pgReleaseName string
+		namespace     string
+	)
+
+	BeforeEach(func() {
+		releaseName = fmt.Sprintf("topgun-ep-%d-%d", GinkgoRandomSeed(), GinkgoParallelNode())
+		namespace = releaseName
+		pgReleaseName = releaseName + "-pg"
+
+		helmDeploy(pgReleaseName,
+			namespace,
+			path.Join(Environment.ChartsDir, "stable/postgresql"),
+			"--set=livenessProbe.initialDelaySeconds=3",
+			"--set=livenessProbe.periodSeconds=3",
+			"--set=persistence.enabled=false",
+			"--set=postgresqlDatabase=pg-database",
+			"--set=postgresqlPassword=pg-password",
+			"--set=postgresqlUsername=pg-user",
+			"--set=readinessProbe.initialDelaySeconds=3",
+			"--set=readinessProbe.periodSeconds=3",
+		)
+
+		deployConcourseChart(releaseName,
+			"--set=concourse.web.postgres.database=pg-database",
+			"--set=concourse.web.postgres.host="+pgReleaseName+"-postgresql",
+			"--set=concourse.worker.ephemeral=true",
+			"--set=postgresql.enabled=false",
+			"--set=secrets.postgresPassword=pg-password",
+			"--set=secrets.postgresUser=pg-user",
+			"--set=worker.replicas=0",
+		)
+	})
+
+	AfterEach(func() {
+		helmDestroy(releaseName)
+		helmDestroy(pgReleaseName)
+		Wait(Start(nil, "kubectl", "delete", "namespace", namespace, "--wait=false"))
+	})
+
+	It("gets deployed", func() {
+		waitAllPodsInNamespaceToBeReady(namespace)
```

Comment:
I am not sure, should we add more test to make sure that interaction with the db is working correctly? 
like setting a pipeline and getting the pipeline config?

Reply:
Hey, that's seems like a pretty valid concern; my rationale was around the fact that `web` won't get ready (per the [`readinessProbe`](https://github.com/helm/charts/blob/7e9730490ac0aacabfe5236e60dec3f7b2653051/stable/concourse/values.yaml#L1124-L1127)) before all connectivity gets established (which includes performing the migrations as no endpoints will respond before that)  🤔 

To be more thorough then, should we:

- [’logging in’](https://github.com/concourse/concourse/blob/bf12d2f50c6b5d940bb80b60299f30bb10185f2d/topgun/k8s/kubernetes_creds_mgmt_test.go#L36-L37), and then
- [`set-pipeline`](https://github.com/concourse/concourse/blob/bf12d2f50c6b5d940bb80b60299f30bb10185f2d/topgun/k8s/kubernetes_creds_mgmt_test.go#L81)?

As I see, if nothing errors on such path, we should be good.

Wdyt?

thx!

Reply:
Yeah, I didn't really think about the migrations. Maybe we don't need that?

Reply:
👍 I'll include an extra step then!